pkg/controller/kibana/driver.go

```diff
@@ -233,9 +233,7 @@ func (d *driver) Reconcile(
 	params operator.Parameters,
 ) *reconciler.Results {
 	results := reconciler.Results{}
-	if kb.RequiresAssociation() && !kb.AssociationConf().IsConfigured() {
-		d.recorder.Event(kb, corev1.EventTypeWarning, events.EventAssociationError, "Elasticsearch backend is not configured")
-		log.Info("Elasticsearch association not established: skipping Kibana deployment reconciliation", "namespace", kb.Namespace, "kibana_name", kb.Name)
+	if !association.IsConfigured(kb, d.recorder) {
```

Comment:
Is there any particular reason for putting this code here instead of in the `doReconcile` method of the kibana controller?

Reply:
It is its original place 🤷‍♂ 
But I can move it.

Reply:
I moved the check for the APM association before the `doReconcile`, I guess it makes sense since it is a precondition for the actual reconciliation loop.
Let me know what you think.

Reply:
Looks good.